src/components/pages/donation_form/AddressTypeBasic.vue

```diff
@@ -1,108 +1,91 @@
+        {{ $t( 'donation_form_address_choice_direct_debit_disclaimer_addresstype_basic' ) }}
+      </div>
+    </template>
+  </RadioField>
 
-      <RadioInput
-          name="addressType"
-          v-model="addressType"
-          native-value="person"
-      >{{ $t( 'donation_form_addresstype_option_private_addresstype_basic' ) }}
-      </RadioInput>
-
-      <RadioInput
-          name="addressType"
-          v-model="addressType"
-          native-value="company"
-      >
-        {{ $t( 'donation_form_addresstype_option_company_addresstype_basic' ) }}
-      </RadioInput>
-
-      <RadioInput
-          id="anonymous"
-          name="addressType"
-          v-model="addressType"
-          native-value="anonymous"
-          :disabled="disableAnonymous">
-        {{ $t( 'donation_form_addresstype_option_anonymous_addresstype_basic' ) }}
-      </RadioInput>
-
-		</div>
-
-	</fieldset>
 </template>
 
-<script lang="ts">
-import { addressTypeFromName, AddressTypeModel } from '@src/view_models/AddressTypeModel';
-import { computed, defineComponent, PropType, Ref, ref, watch } from 'vue';
-import RadioInput from '@src/components/shared/legacy_form_inputs/RadioInput.vue';
+<script setup lang="ts">
 
-export default defineComponent( {
-	name: 'AddressTypeBasic',
-	components: { RadioInput },
-	props: {
-		disabledAddressTypes: Array as PropType<Array<AddressTypeModel>>,
-		initialAddressType: String,
-		isDirectDebit: Boolean,
-	},
-	setup( props, { emit } ) {
-
-		const addressType = ref( '' );
+import RadioField from '@src/components/shared/form_fields/RadioField.vue';
+import { AddressTypeModel } from '@src/view_models/AddressTypeModel';
+import { computed, ref, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
 
-		const initialAddressTypeString = props.initialAddressType ? props.initialAddressType : 'unset';
+type AddressStates = 'anonymous' | 'person' | 'company' | 'unset';
 
-		switch ( addressTypeFromName( initialAddressTypeString ) ) {
-			case AddressTypeModel.ANON:
-				addressType.value = 'anonymous';
-				break;
-			case AddressTypeModel.UNSET:
-				addressType.value = initialAddressTypeString;
-				break;
-			case AddressTypeModel.PERSON:
-				addressType.value = 'person';
-				break;
-			case AddressTypeModel.COMPANY:
-				addressType.value = 'company';
-				break;
-		}
-		const type: Ref<AddressTypeModel> = ref( addressTypeFromName( initialAddressTypeString ) );
+interface Props {
+	disabledAddressTypes: AddressTypeModel[]
+	isDirectDebit: boolean;
+  addressTypeIsInvalid: boolean;
+	initialAddressType?: AddressStates;
+}
+const props = defineProps<Props>();
 
-		emit( 'set-full-selected', true );
+const emit = defineEmits( [ 'address-type', 'set-full-selected' ] );
 
-		const disableAnonymous = computed( (): boolean => props.disabledAddressTypes !== undefined && props.disabledAddressTypes.includes( AddressTypeModel.ANON ) );
+const addressType = ref<AddressStates>( props.initialAddressType ?? 'unset' );
```

Comment:
You're losing the address type on reload because the initial address type is coming in as `firma` and the field is looking for `company`

Reply:
I would suggest using the address type integer values as the field values. That will remove the requirement to convert text at all. That's how I handled the field in the new address form.